Description: Include a copy of QAbstractDynamicMetaObject in qmetadatacontrolmetaobject_p.h

QMetaDataControlMetaObject inherits from QAbstractDynamicMetaObject but the
qobject_p.h header can't be included when building against packages which don't
include private headers. We have to make do with a copy of the class instead.

Author: Fathi Boudra <fabo@debian.org>

--- a/plugins/declarative/multimedia/qmetadatacontrolmetaobject_p.h
+++ b/plugins/declarative/multimedia/qmetadatacontrolmetaobject_p.h
@@ -59,15 +59,27 @@
 // We mean it.
 //
 
-#include <QtCore/qmetaobject.h>
+#include <QtCore/QMetaObject>
+#include <QtCore/QObject>
 #include <qtmedianamespace.h>
 
-#include <QtCore/private/qobject_p.h>
-
 QT_BEGIN_HEADER
 
 QT_BEGIN_NAMESPACE
 
+#ifndef QOBJECT_P_H
+#if (QT_VERSION >= QT_VERSION_CHECK(4, 6, 0))
+
+struct Q_CORE_EXPORT QAbstractDynamicMetaObject : public QMetaObject
+{
+    virtual ~QAbstractDynamicMetaObject() {}
+    virtual int metaCall(QMetaObject::Call, int _id, void **) { return _id; }
+    virtual int createProperty(const char *, const char *) { return -1; }
+};
+
+#endif
+#endif
+
 class QMetaDataReaderControl;
 
 class QMetaDataControlMetaObject : public QAbstractDynamicMetaObject
